include/seqan3/io/alignment_file/format_sam.hpp

```diff
@@ -162,6 +169,239 @@ class alignment_file_format_sam
+
+        // Fields 1-5: ID FLAG REF_ID REF_OFFSET MAPQ
+        // -------------------------------------------------------------------------------------------------------------
+        read_field(stream_view | next, id);
+        ++ranges::begin(stream_view); // skip tab
+
+        read_field(stream_view | next, flag);
+        ++ranges::begin(stream_view);
+
+        read_field(stream_view | next, ref_id);
+        ++ranges::begin(stream_view);
+
+        read_field(stream_view | next, ref_offset_tmp);
+        ++ranges::begin(stream_view);
+        --ref_offset_tmp; // SAM format is 1-based but SeqAn operates 0-based.
+        if constexpr (!detail::decays_to_ignore_v<ref_offset_type>)
+            ref_offset = ref_offset_tmp;
+
+        read_field(stream_view | next, mapq);
+        ++ranges::begin(stream_view);
+
+        // Field 6: CIGAR
+        // -------------------------------------------------------------------------------------------------------------
+        if constexpr (!detail::decays_to_ignore_v<align_type>)
+        {
+            cigar = parse_cigar(offset_tmp, soft_clipping_end, stream_view | next);
+
+            // Compute length of aligned reference and query sequence in one go.
+            // The length of query sequence is needed when parsing the sequence (Field 10)
+            for (auto [cigar_op, cigar_count] : cigar)
+            {
+                if (is_char<'M'>(cigar_op))
+                {
+                    ref_length += cigar_count;
+                    seq_length += cigar_count;
+                }
+                else if (is_char<'D'>(cigar_op))
+                {
+                    ref_length += cigar_count;
+                }
+                else if (is_char<'I'>(cigar_op))
+                {
+                    seq_length += cigar_count;
+                }
+            }
+
+            if constexpr (!detail::decays_to_ignore_v<offset_type>)
+            {
+                offset = offset_tmp;
+            }
+        }
+        else
+        {
+            detail::consume(stream_view | next);
+        }
+        ++ranges::begin(stream_view);
+
+        // Field 7-9: (RNEXT PNEXT TLEN)=MATE
+        // -------------------------------------------------------------------------------------------------------------
+        if constexpr (!detail::decays_to_ignore_v<mate_type>)
+        {
+            read_field(stream_view | next, get<0>(mate)); // RNEXT
+            ++ranges::begin(stream_view);
+
+            read_field(stream_view | next, get<1>(mate)); // PNEXT
+            ++ranges::begin(stream_view);
+
+            read_field(stream_view | next, get<2>(mate)); // TLEN
+            ++ranges::begin(stream_view);
+        }
+        else
+        {
+            for (size_t i = 0; i < 3; ++i)
+            {
+                read_field(stream_view | next, std::ignore);
+                ++ranges::begin(stream_view);
+            }
```

Comment:
either consume (which might be faster than read_field), or don't do `if constexpr`, bceause you can assign to `std::ignore`